src/NuGet.Core/NuGet.Protocol/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -56,6 +58,11 @@ private HttpHandlerResourceV3 CreateResource(PackageSource packageSource)
                 AutomaticDecompression = (DecompressionMethods.GZip | DecompressionMethods.Deflate),
             };
 
+            if (packageSource.DisableTLSCertificateValidation)
```

Comment:
I think we should issue a warning if customers set this property, stating that this setting is not a secure option in a production environment.

Reply:
Added a warning message.

Reply:
Given that this is a design change, I think this should be covered in the spec as well. 

I'm wondering if adding a warning for something customers explicitly disabled is a bit too loud. 

The workflow here is:
- You get an HTTP deprecation warning.
- You're told about deprecation options, you set the disable tls flag.
- Restore again, get a warning about the tls flag that you just set as a response to the deprecation warning. 

Some considerations about warnings being raised in this codepath: 
- How often is the warning raised per source?
- Does every project get the warning or just one?